// @vx/shape/lib/shapes/SplitLinePath.js
"use strict";

exports.__esModule = true;
exports.default = SplitLinePath;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _LinePath = _interopRequireDefault(require("./LinePath"));

var _getLineSegments = _interopRequireDefault(require("./getLineSegments"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function SplitLinePath(_ref) {
  var children = _ref.children,
      curve = _ref.curve,
      segments = _ref.data,
      defined = _ref.defined,
      x = _ref.x,
      y = _ref.y,
      styles = _ref.styles;
  // combine data to first draw entire path
  var flatData = segments.reduce(function (flat, segmentData) {
    return flat.concat([].concat(segmentData));
  }, []);
  return _react.default.createElement(_LinePath.default, {
    data: flatData,
    defined: defined,
    curve: curve,
    x: x,
    y: y
  }, function (_ref2) {
    var path = _ref2.path;
    var wholePath = path(flatData);
    var computedLineSegments = (0, _getLineSegments.default)(wholePath || '', segments);
    return computedLineSegments.map(function (segment, i) {
      return _react.default.createElement(_react.default.Fragment, {
        key: i
      }, children ? children({
        index: i,
        segment: segment,
        styles: styles[i] || styles[i % styles.length]
      }) : _react.default.createElement(_LinePath.default, _extends({
        key: i,
        data: segment,
        x: function x(d) {
          return d.x || 0;
        },
        y: function y(d) {
          return d.y || 0;
        }
      }, styles[i] || styles[i % styles.length])));
    });
  });
}

SplitLinePath.propTypes = {
  data: _propTypes.default.arrayOf(_propTypes.default.arrayOf(_propTypes.default.any)).isRequired,
  styles: _propTypes.default.arrayOf(_propTypes.default.any).isRequired,
  curve: _propTypes.default.any,
  defined: _propTypes.default.func,
  x: _propTypes.default.func,
  y: _propTypes.default.func,
  children: _propTypes.default.func
};
