// @vx/shape/lib/shapes/SplitPath.js
"use strict";

exports.__esModule = true;
exports.default = SplitPath;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _LinePath = _interopRequireDefault(require("./LinePath"));

var _getLineSegments = _interopRequireDefault(require("./getLineSegments"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function SplitPath(_ref) {
  var curve = _ref.curve,
      segments = _ref.data,
      defined = _ref.defined,
      x = _ref.x,
      y = _ref.y;
  // combine data to first draw entire path
  var flatData = segments.reduce(function (flat, segmentData) {
    return flat.concat([].concat(segmentData));
  }, []);
  return _react.default.createElement(_LinePath.default, {
    data: flatData,
    defined: defined,
    curve: curve,
    x: x,
    y: y
  }, function (_ref2) {
    var path = _ref2.path;
    var wholePathDAttribute = path(flatData);
    var computedLineSegments = (0, _getLineSegments.default)(wholePathDAttribute || '', segments);
    return computedLineSegments.map(function (segment, i) {
      return _react.default.createElement(_LinePath.default, {
        key: i,
        data: segment,
        x: function x(d) {
          return d.x;
        },
        y: function y(d) {
          return d.y;
        }
      });
    });
  });
}

SplitPath.propTypes = {
  data: _propTypes.default.arrayOf(_propTypes.default.arrayOf(_propTypes.default.any)).isRequired,
  styles: _propTypes.default.arrayOf(_propTypes.default.any).isRequired,
  curve: _propTypes.default.any,
  defined: _propTypes.default.func,
  x: _propTypes.default.func,
  y: _propTypes.default.func
};
